data/every_dream.py

```diff
@@ -16,108 +16,61 @@
         self.retain_contrast = retain_contrast
-        self.write_schedule = write_schedule
         self.shuffle_tags = shuffle_tags
         self.seed = seed
         self.rated_dataset = rated_dataset
         self.rated_dataset_dropout_target = rated_dataset_dropout_target
-
-        if seed == -1:
-            seed = random.randint(0, 99999)
-        
-        if not dls.shared_dataloader:
-            logging.info(" * Creating new dataloader singleton")
-            dls.shared_dataloader = dlma(data_root=data_root,
-                                         seed=seed,
-                                         debug_level=debug_level,
-                                         batch_size=self.batch_size,
-                                         flip_p=flip_p,
-                                         resolution=resolution,
-                                         log_folder=self.log_folder,
-                                        )
-
-        self.image_train_items = dls.shared_dataloader.get_shuffled_image_buckets(1.0) # First epoch always trains on all images
-
+        # First epoch always trains on all images
+        self.image_train_items = self.data_loader.get_shuffled_image_buckets(1.0)
+            
         num_images = len(self.image_train_items)
-
-        logging.info(f" ** Trainer Set: {num_images / batch_size:.0f}, num_images: {num_images}, batch_size: {self.batch_size}")
-        if self.write_schedule:
-            self.__write_batch_schedule(0)
-
-    def __write_batch_schedule(self, epoch_n):
-        with open(f"{self.log_folder}/ep{epoch_n}_batch_schedule.txt", "w", encoding='utf-8') as f:
-            for i in range(len(self.image_train_items)):
-                try:
-                    f.write(f"step:{int(i / self.batch_size):05}, wh:{self.image_train_items[i].target_wh}, r:{self.image_train_items[i].runt_size}, path:{self.image_train_items[i].pathname}\n")
-                except Exception as e:
-                    logging.error(f" * Error writing to batch schedule for file path: {self.image_train_items[i].pathname}")
-
-    def get_runts():
-        return dls.shared_dataloader.runts
+        logging.info(f" ** Trainer Set: {num_images / self.batch_size:.0f}, num_images: {num_images}, batch_size: {self.batch_size}")
 
     def shuffle(self, epoch_n: int, max_epochs: int):
         self.seed += 1
-        if dls.shared_dataloader:
-            if self.rated_dataset:
-                dropout_fraction = (max_epochs - (epoch_n * self.rated_dataset_dropout_target)) / max_epochs
-            else:
-                dropout_fraction = 1.0
-
-            self.image_train_items = dls.shared_dataloader.get_shuffled_image_buckets(dropout_fraction)
+        
+        if self.rated_dataset:
+            dropout_fraction = (max_epochs - (epoch_n * self.rated_dataset_dropout_target)) / max_epochs
         else:
-            raise Exception("No dataloader singleton to shuffle")
```

Comment:
I don't think `dls` is needed anymore with DLMA being passed in.

Reply:
It was actually only there as a leftover from ED1 because Lightning would instantiate the dataset class 4 or 6 times (and in my case for ED1 would do preloading/prescan that many times as well).  It was never actually needed in ED2 just a leftover.